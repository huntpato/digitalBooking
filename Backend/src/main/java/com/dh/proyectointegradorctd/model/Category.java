package com.dh.proyectointegradorctd.model;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "categories")
public class Category {

    /*============= Atributos =============== */

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;
    private String name;
    private String photoUrl;
    private String description;
    private String countProducts;

    /* ============= Getters y Setters ============== */

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public void setPhotoUrl(String photoUrl) {
        this.photoUrl = photoUrl;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCountProducts() {
        return countProducts;
    }

    public void setCountProducts(String countProducts) {
        this.countProducts = countProducts;
    }

    /* ================= Constructor ==================== */

    public Category(String name, String photoUrl, String description) {
        this.name = name;
        this.photoUrl = photoUrl;
        this.description = description;
    }

    public Category() {
    }
}
