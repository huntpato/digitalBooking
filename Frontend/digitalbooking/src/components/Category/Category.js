import React from "react";
import { ProductsByCategory } from "../ProductsByCategory/ProductsByCategory";
import { useLocation, useHistory } from 'react-router-dom';
import useFetch from "use-http";
import styles from "./Category.module.css";

export const Category = () => {
  const location = useLocation();
  const path= location.pathname.split("/");
  const id = path[2];
  const options = { headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const url = `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/category/${path[2]}`;
  const { loading, data = [] } = useFetch(url, options, []);
  const { category, title, cat, titleContainer } = styles;
  let history = useHistory();
  
  
  return (
    <div className={cat}>
        <section className= {title}>
          <div className={titleContainer}>
            <h4 className= {category}>{data.name}</h4>
            <i className="fas fa-arrow-left" onClick={() => history.goBack()}></i>
          </div>
        </section>
          {loading && <p>Cargando Recomendaciones...</p>}  
          <ProductsByCategory category={id} name={data.name}/>
    </div>
  );
};