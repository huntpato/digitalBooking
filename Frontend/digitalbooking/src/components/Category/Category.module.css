.cat{
  min-height: 93vh;
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  background-color: var(--background-color2);
  padding: 68px 0px 0px 0px;
}

.title{
  align-items: center;
  background-color: var(--color-typo);
  display:flex;
  flex-direction: column;
  justify-content: center;
  padding: 30px 0px 0px 0px;
}

.titleContainer{
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 20px 0px 10px 0px;
  margin-bottom: 15px;
  width: 90vw;
}

.title i {
  color: white;
  padding: 0px;
}

.category{
  color: var(--background-white);
  font-size: 22px;
  
}

@media (max-width: 768px) {
  .category {
    /**/ 
  }

  .recommendation h4 {
    margin-bottom: 10px;
    margin-left: 0px;
  }

  .recommendationCards {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}

@media (max-width: 480px) {
}

@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

