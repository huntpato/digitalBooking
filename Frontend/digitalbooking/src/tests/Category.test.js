import React from "react";
import { shallow } from "enzyme";
import { Category } from "../components/Category/Category";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
  useHistory: jest.fn(),
  useLocation:jest.fn().mockReturnValue({
    pathname: "/register",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
}));

describe("Pruebas en componente <Category />", () => {
  const wrapper = shallow(<Category />);

  test("Comprobar que <Category /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Verificar que el componente carga la clase de CSS correctamente", () => {
    const div = wrapper.find("div").at(0);
    expect(div.hasClass("cat")).toBe(true);
  });
});
