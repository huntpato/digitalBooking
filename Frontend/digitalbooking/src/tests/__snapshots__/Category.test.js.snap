// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Category /> Comprobar que <Category /> renderiza correctamente. 1`] = `
<div
  className="cat"
>
  <section
    className="title"
  >
    <div
      className="titleContainer"
    >
      <h4
        className="category"
      />
      <i
        className="fas fa-arrow-left"
        onClick={[Function]}
      />
    </div>
  </section>
  <p>
    Cargando Recomendaciones...
  </p>
  <ProductsByCategory />
</div>
`;
